client/dashboard/sites/overview/test/storage-warning-banner.tsx

```diff
@@ -0,0 +1,116 @@
+/**
+ * @jest-environment jsdom
+ */
+import { screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import nock from 'nock';
+import { render } from '../../../test-utils';
+import { StorageWarningBanner } from '../storage-warning-banner';
+
+interface TestOptions {
+	storage_used_bytes: number;
+	max_storage_bytes: number;
+	isDismissed?: boolean | string;
+}
+
+const siteId = 123;
+
+// Only mocks site and settings fields that are necessary for the tests.
+// Feel free to add more fields as they are needed.
+function mockTestEndpoints( options: TestOptions ) {
+	const { storage_used_bytes, max_storage_bytes, isDismissed = false } = options;
+
+	const mediaStorage = {
+		storage_used_bytes,
+		max_storage_bytes,
+		max_storage_bytes_from_addons: 0,
+	};
+
+	const preferences = {
+		[ `hosting-dashboard-overview-storage-notice-dismissed-${ siteId }` ]:
+			typeof isDismissed === 'string' ? isDismissed : isDismissed && 'today',
+	};
+
+	const scope = nock( 'https://public-api.wordpress.com' )
```

Comment:
[nit] Maybe we can try msw as it looks better and has better integration with the storybook 😄

Reply:
I saw launchpad and plan-grid-next use it for the storybook.